Bugfix.
1) the string nstore might overflow
2) strncpy does not always nul terminate a string


--- ./v1.10.src/XC16_v1_10_branch/gcc/gcc/gcc.c.orig	2011-01-24 13:49:37.000000000 +0100
+++ ./v1.10.src/XC16_v1_10_branch/gcc/gcc/gcc.c	2011-01-24 14:48:51.000000000 +0100
@@ -3412,7 +3412,7 @@
 #endif
         {
           const char *startp, *endp;
-          char *nstore = (char *) alloca (strlen (temp) + 3);
+          char *nstore = (char *) alloca (strlen (temp) + 3 + strlen(base_path) + strlen(dir_separator_str));
 
           startp = endp = temp;
           while (1)
@@ -3423,6 +3423,7 @@
                     strcpy (nstore, concat (".", dir_separator_str, NULL));
                   else if (IS_DIR_SEPARATOR(*startp)) {
                     strncpy (nstore, startp, endp - startp);
+		    nstore[endp-startp] = 0;
                   } else {
                     nstore[0] = 0;
                     strcat(nstore, base_path);
